pyispyb/core/routes/webservices/userportalsync.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from fastapi import HTTPException, status, Depends
+from fastapi.responses import Response, JSONResponse
+from pyispyb.app.base import AuthenticatedAPIRouter
+from ...modules import userportalsync as crud
+from ...schemas import userportalsync as schema
+from ....dependencies import permission
+
+
+logger = logging.getLogger("ispyb")
+
+router = AuthenticatedAPIRouter(prefix="/userportalsync", tags=["User Portal Sync"])
+
+
+@router.post(
+    "/sync_proposal",
+    response_model=schema.UserPortalProposalSync,
+)
+def sync_proposal(
+    proposal: schema.UserPortalProposalSync,
+    depends: bool = Depends(permission("uportal_sync")),
+):
+    """Create/Update a proposal from the User Portal and all its related entities"""
+    try:
+        execution_time = crud.sync_proposal(proposal=proposal)
+        proposal_dict = proposal.dict()
+        return JSONResponse(
```

Comment:
Your response_model, and actual output seem to be a bit confused here. You dont seem to actually respond with a `schema.UserPortalProposalSync`, you look to be sending a message.

You dont need a JSONResponse, just send a python dictionary, the default status_code is 200 so thats not needed either.

You could define a new reusable `Message` model in `core/routes/responses.py`, e.g.
```
class Message(BaseModel):
    message: str
```

then `repsonse_model` for this function becomes `Message`

Reply:
Done!

https://github.com/ispyb/py-ispyb/pull/210/commits/7593bc6be69ed08a33cc0dbb0d28c39a22329890